### PR TITLE
Add support for Base58 and Multibase key formats

### PR DESCRIPTION
**Pull request by spivachuk**

### CODE REVIEW

File: lib/src/main/kotlin/org/didcommx/didcomm/exceptions/Exceptions.kt [hunk 1]
```diff
@@ -9,11 +12,42 @@ package org.didcommx.didcomm.exceptions
 open class DIDCommException(message: String, cause: Throwable? = null) : Throwable(message, cause)
 
 /**
- * This exception SHOULD be raised if secret can not be converted to JWK format.
+ * This exception SHOULD be raised if verification method type is not supported.
+ *
+ * @param type The verification method type.
+ */
+class UnsupportedVerificationMethodTypeException(type: VerificationMethodType) :
```
Comment (ashcherbakov) [inline lib/src/main/kotlin/org/didcommx/didcomm/exceptions/Exceptions.kt:19]: I think we have too many similar Unsupported exceptions. Most probably it doesn't make sense for the user to catch so many, and it's hard to say what exception to catch.
I would make just one DIDCommUnsupportedException, and put the details about what is unsupported into the message itself. We can put there an optional error code as well.

File: lib/build.gradle [hunk 2]
```diff
@@ -23,6 +23,7 @@ kotlin {
 
 dependencies {
     implementation "com.nimbusds:nimbus-jose-jwt:${nimbusJoseJWTVersion}"
+    implementation "com.github.multiformats:java-multibase:${javaMultibase}"
```
Comment (ashcherbakov) [inline lib/build.gradle:26]: Please add 
`include(dependency("com.github.multiformats:java-multibase"))` into `dependencies` of `shadowJar`
(see https://github.com/sicpa-dlab/peer-did-jvm/blob/main/lib/build.gradle#L82)

File: lib/src/main/kotlin/org/didcommx/didcomm/crypto/key/Key.kt [hunk 3]
```diff
@@ -49,14 +100,57 @@ sealed interface Key {
...
+
+        override lateinit var curve: Curve
+            private set
+
+        init {
+            val rawValue = Base58.decode(materialValue)
+            val base64URLValue = Base64URL.encode(rawValue).toString()
+
+            val jwkJson: Map<String, Any> = mapOf(
+                "kty" to "OKP",
```
Comment (ashcherbakov) [inline lib/src/main/kotlin/org/didcommx/didcomm/crypto/key/Key.kt:115]: Should we add `kid` (equal to `id` field) to the JWK?

File: lib/src/main/kotlin/org/didcommx/didcomm/exceptions/Exceptions.kt (hunk 1 above)
Comment (spivachuk) [reply lib/src/main/kotlin/org/didcommx/didcomm/exceptions/Exceptions.kt:26]: Added `DIDCommUnsupportedValueException` base class.

**Pull request merged**